packages/back-end/src/scim/groups/listGroups.ts

```diff
@@ -0,0 +1,72 @@
+import { parse, filter } from "scim2-parse-filter";
+import { getTeamsForOrganization } from "../../models/TeamModel";
+import { expandOrgMembers } from "../../services/organizations";
+import { createApiRequestHandler } from "../../util/handler";
+import { COUNT_DEFAULT, START_INDEX_DEFAULT } from "../users/listUsers";
+import {
+  ScimGroup,
+  ScimListRequest,
+  ScimListResponse,
+} from "../../../types/scim";
+
+export const listGroups = createApiRequestHandler()(
+  async (req: ScimListRequest): Promise<ScimListResponse> => {
+    const { startIndex, count, filter: filterQuery } = req.query;
+
+    const queryOptions = {
+      startIndex: startIndex
+        ? parseInt(startIndex as string)
+        : START_INDEX_DEFAULT,
+      count: count ? parseInt(count as string) : COUNT_DEFAULT,
+    };
+
+    const org = req.organization;
+
+    const groups = await getTeamsForOrganization(org.id);
+
+    const groupsWithMembersP = groups.map(async (group) => {
+      const members = org.members.filter((member) =>
+        member.teams?.includes(group.id)
+      );
+      const expandedMembers = await expandOrgMembers(members);
+      return {
+        ...group,
+        members: expandedMembers,
+      };
+    });
```

Comment:
Not important for this PR, but I imagine in large orgs, it will be very common for a single user to be in multiple groups.  Would be more efficient to build a user -> expandedUser mapping once and then hydrate all the groups synchronously.